source/MRMesh/MRSceneRoot.cpp

```diff
@@ -30,23 +32,46 @@ SceneRoot& SceneRoot::instace_()
 
 SceneRoot::SceneRoot()
 {
-    root_ = std::make_shared<Object>();
+    root_ = std::make_shared<SceneRootObject>();
     root_->setName( "Root" );
 }
 
-const Object& SceneRoot::constGet()
+const SceneRootObject& SceneRoot::constGet()
 {
     return *instace_().root_;
 }
 
-std::shared_ptr<const Object> SceneRoot::constGetSharedPtr()
+std::shared_ptr<const SceneRootObject> SceneRoot::constGetSharedPtr()
 {
-    return std::const_pointer_cast<const Object>( instace_().root_ ); 
+    return std::const_pointer_cast<const SceneRootObject >( instace_().root_ );
 }
 
 const std::filesystem::path& SceneRoot::getScenePath()
 {
     return instace_().scenePath_;
 }
 
+MR_ADD_CLASS_FACTORY( SceneRootObject )
+
+SceneRootObject::SceneRootObject()
+{
+    setName( "Root" );
+    setAncillary( false );
+}
+
+void SceneRootObject::serializeFields_( Json::Value& root ) const
+{
+    Object::serializeFields_( root );
+    // append base type
+    root["Type"].append( SceneRootObject::TypeName() );
+}
+
+void SceneRootObject::deserializeFields_( const Json::Value& root )
+{
+    Object::deserializeFields_( root );
+    Object::select( false );
```

Comment:
A comment above the line to state that these changes are required for scene root objects.